[JENKINS-71177] modal with title for legend

mawinter69 opened the pull request:
The modal feature is displaying to complete content inside a div. For the legend which is rather big the content must be scrolled unless you're on a bigger screen. At present this lead to the content being displayed without any margins and the scrollbar arrows at top and bottom are partially overlapping with the rounded corners of the modal.

This change introduces a dedicated title for the modal. The title will be fixed on top of the modal and only the content is scrollable. A proper margin at the bottom ensures that the scrollbar is not overlapping with the corners.

Before:
![image](https://user-images.githubusercontent.com/17767050/235679679-b05075ef-b714-48b9-b617-a1977d3bfc99.png)

After::
![image](https://user-images.githubusercontent.com/17767050/235687428-60b08709-c3f3-4476-80ad-f0c83fa6648a.png)

<!-- Comment:
A great PR typically begins with the line below.
Replace XXXXX with the numeric part of the issue ID you created in Jira.
Note that if you want your changes backported into LTS, you need to create a Jira issue. See https://www.jenkins.io/download/lts/#backporting-process for more information.
-->

See [JENKINS-71177](https://issues.jenkins.io/browse/JENKINS-71177).

<!-- Comment:
If the issue is not fully described in Jira, add more information here (justification, pull request links, etc.).

 * We do not require Jira issues for minor improvements.
 * Bug fixes should have a Jira issue to facilitate the backporting process.
 * Major new features should have a Jira issue.
-->

### Testing done
Tested manually, no Java code changes

<!-- Comment:
Provide a clear description of how this change was tested.
At minimum this should include proof that a computer has executed the changed lines.
Ideally this should include an automated test or an explanation as to why this change has no tests.
Note that automated test coverage is less than complete, so a successful PR build does not necessarily imply that a computer has executed the changed lines.
If automated test coverage does not exist for the lines you are changing, you must describe the scenario(s) in which you manually tested the change.
For frontend changes, include screenshots of the relevant page(s) before and after the change.
For refactoring and code cleanup changes, exercise the code before and after the change and verify the behavior remains the same.
-->

### Proposed changelog entries

- Make title sticky in legend.

<!-- Comment:
The changelog entry should be in the imperative mood; e.g., write "do this"/"return that" rather than "does this"/"returns that".
For examples, see: https://www.jenkins.io/changelog/
-->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/core-pr-reviewers.
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).


<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7922"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

mawinter69 committed:
modal with title for legend

The modal feature is displaying to complete content inside a div. For
the legend which is rather big the content must be scrolled. At present
this lead to the content being displayed without any margins and the
scrollbar arrows at top and bottom are partially overlapping with the
rounded corners of the modal.

This change introduces a dedicated title for the modal. The title will
be fixed on top of the modal and only the content is scrollable. A
proper margin at the bottom ensures that the scrollbar is not
overlapping with the corners.

timja reviewed (approved):
nice

janfaracik commented:
Thanks!

Couple of questions -

+1 for making the title an option for showModal. How would you feel about moving it inside of the options {} parameter?

How would you feel about reducing the corner radius instead of the margin at the bottom? It looks a bit odd the content cutting off at the bottom.

mawinter69 committed:
make the title optional

reduce corner radius and remove the margin

mawinter69 commented:
> Thanks!
> 
> Couple of questions -
> 
> +1 for making the title an option for showModal. How would you feel about moving it inside of the options {} parameter?
> 
> How would you feel about reducing the corner radius instead of the margin at the bottom? It looks a bit odd the content cutting off at the bottom.

done
<img width="422" alt="image" src="https://user-images.githubusercontent.com/17767050/235989750-1f75a47a-d84e-4305-adb5-925ed3cdf22b.png">

NotMyFault reviewed (approved):
Nice

mawinter69 committed:
minor bug fixes

the rssbar_de change was unnecessary

the closeButton is not null but undefined when hidden

NotMyFault commented:
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!
